api_app/api.py

```diff
@@ -32,6 +33,77 @@
 logger = logging.getLogger(__name__)
 
 
+def send_for_analysis(prep_data, test):
+    """
+    Send task (file/observable) for analysis
+    """
+
+    # Check if task is test or not
+    if not test:
+        # fire celery task
+        celery_app.send_task(
+            "start_analyzers",
+            kwargs=dict(
+                job_id=prep_data["job_id"],
+                analyzers_to_execute=prep_data["analyzers_to_execute"],
+                runtime_configuration=prep_data["runtime_config"],
+            ),
+        )
+
+    response_dict = {
+        "status": "accepted",
+        "job_id": prep_data["job_id"],
+        "analyzers_running": prep_data["analyzers_to_execute"],
+    }
+
+    logger.debug(response_dict)
+
+    return response_dict
+
+
+def prepare_for_analysis(serializer, params, warnings):
```

Comment:
While this approach with 2 common functions `send_for_analysis` and `prepare_for_anlaysis` is okay, there's a better way I just thought of:

Combine these 2 functions into an individual function 

```python3
def _analysis_request(request, serializer_class):
```

And then the `analyze_file` and `analyze_observable` would look like:

```python3
@api_view()
def analyze_file(request):
    return _analysis_request(request, FileAnalysisSerializer)

@api_view()
def analyze_observable(request):
    return _analysis_request(request, ObservableAnalysisSerializer)
```